### PR TITLE
Not so local RPC client

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

#2909.

### Solution

Instead of RPC server rework we just provide a (wrapped) WSClient that does calls in the same process. RPC server rework is something that can be done, but:
 * RPC client and RPC server interfaces do not exactly match in the way they handle inputs and outputs
 * we'll have to manage concurrency in this client (and manage it again in the RPC server for subscriptions!)

This structure is better in that:
 * for `Internal` user it's absolutely the same interface as `WSClient` has, they can be interchanged easily and we can have this link configurable (go via RPC or use internal)
 * it has much simpler implementation
 * subscriptions are much easier with `WSClient` interface (and more robust, RPC server has some buffers and already manages queue overflows)

But:
 * it adds a little JSON in/out overhead
 * it leaves us with a lot of code still in the RPC server (but hey, it works)

Not yet complete.

### CODE REVIEW

**Comment by roman-khimov** [comment]
@cthulhu-rider, check it out. Especially `TestLocalClient` in the `rpcsrv`, it shows that the RPC server could be started with zero ports exposed and local `Internal` client can happily use it. It at the same time is 100% interchangeable with `WSClient`. So the idea is that you keep RPC server config internally (except maybe for `MaxGasInvoke`, but you enable sessions, do any other things needed), it's not exposed, yet at the same time its API works as usual.

File: pkg/services/rpcsrv/server.go [hunk 1]
```diff
@@ -518,6 +531,51 @@ func (s *Server) handleRequest(req *params.Request, sub *subscriber) abstractRes
...
+				},
+			},
+		}
+	)
+	reqParams, err := params.FromAny(req.Params)
+	if err != nil {
+		return nil, err
+	}
+
+	s.log.Debug("processing rpc request",
```
Comment (AnnaShaleva) [inline pkg/services/rpcsrv/server.go:552]: Then maybe we can change the log message to something like "processing local rpc request" in order to distinguish simple and local requests?

File: pkg/rpcclient/local.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package rpcclient
+
+import (
+	"context"
+
+	"github.com/nspcc-dev/neo-go/pkg/neorpc"
+	"go.uber.org/atomic"
+)
+
+type InternalHook func(context.Context, chan<- neorpc.Notification) func(*neorpc.Request) (*neorpc.Response, error)
```
Comment (AnnaShaleva) [inline pkg/rpcclient/local.go:10]: Need to be documented.

**Review by AnnaShaleva — COMMENTED** [review]
I like this functionality, don't we want to add a couple of notes about it to the RPC Client documentation?

File: pkg/rpcclient/local.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/rpcclient/local.go:10]: I was expecting this one.

File: pkg/services/rpcsrv/server.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/services/rpcsrv/server.go]: OK.

File: pkg/services/rpcsrv/server.go [hunk 3]
```diff
@@ -547,7 +605,23 @@ func (s *Server) handleIn(req *params.In, sub *subscriber) abstract {
...
-func (s *Server) handleWsWrites(ws *websocket.Conn, resChan <-chan abstractResult, subChan <-chan *websocket.PreparedMessage) {
+func (s *Server) handleLocalNotifications(ctx context.Context, events chan<- neorpc.Notification, subChan <-chan intEvent, subscr *subscriber) {
+eventloop:
+	for {
+		select {
+		case <-s.shutdown:
+			break eventloop
+		case <-ctx.Done():
+			break eventloop
+		case ev := <-subChan:
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/services/rpcsrv/server.go:616]: Fixed.

**Review by AnnaShaleva — APPROVED** [review]
LGTM except the https://github.com/nspcc-dev/neo-go/pull/2916#pullrequestreview-1301526546 comment, but probably method's documentation is enough if it's an internal thing.

**Pull request merged**